src/linker/Linker.Steps/RemoveFiltersStep.cs

```diff
@@ -0,0 +1,76 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+using Mono.Collections.Generic;
+
+namespace Mono.Linker.Steps
+{
+	//
+    // This step removes filter clauses by doing the following transformation:
+    // catch (Exception ex) when (<cond>) {
+    //   ..
+    // }
+    // ->
+    // catch (Exception ex) {
+    //   if (!<cond>)
+    //     throw ex;
+    //     ..
+    //     
+    // }
+	public class RemoveFiltersStep : BaseStep
+    {
+		protected override void Process ()
+        {
+            foreach (var assembly in Context.Annotations.GetAssemblies ()) {
+                RewriteBodies (assembly.MainModule.Types);
```

Comment:
This needs to check if we are linking assembly otherwise do something else (e.g. issue warning).